src/mame/machine/dc_g2if.cpp

```diff
@@ -0,0 +1,538 @@
+// license:LGPL-2.1+
+// copyright-holders:Angelo Salese
+/**************************************************************************************************
+
+    Sega Dreamcast G2 System Bus I/F
+
+    TODO:
+    - Single-step instead of transfering in one go;
+    - Abort DMA if suspend mode is triggered;
+    - Time Out mechanism thru DS# & TR# signals;
+    - External pin enable in trigger select;
+    - Create a pure abstract interface shared with PVR-DMA I/F
+	  (one channel, different max size, simpler tsel, no suspend,
+	   different security code & area protection);
+	- DMA starts should send DDT requests and being notified back to use this i/f implementation
+	  anyway. Exact purpose is unknown, maybe it's for granting use of the bus?
+
+**************************************************************************************************/
+
+#include "emu.h"
+#include "dc_g2if.h"
+
+#define LOG_WARN    (1U << 0)
```

Comment:
You shouldn’t use bit zero for a user-defined log constant – it’s used for `LOG_GENERAL` and the abbreviated `LOG` macro in logmacro.h itself.